File: src/serve.ts
import { MessagingBridge } from "./messaging-bridge";
import type { AgentAdapter, ServeOptions } from "./types";

/**
 * Connect an agent adapter to the Astro messaging service and start listening.
 *
 * This is the framework-agnostic entry point. For Mastra agents, prefer the
 * convenience `serve()` from `@astromode-ai/adapters/mastra` which
 * accepts a Mastra Agent directly.
 */
export function serve(adapter: AgentAdapter, options?: ServeOptions): void {
  const bridge = new MessagingBridge(adapter, options);
  bridge.start().catch((error) => {
    console.error("Fatal error:", error);
    process.exit(1);
  });
}

File: packages/core-py/src/astropods_adapter_core/serve.py
from __future__ import annotations

import asyncio
import json
import logging
import sys
from typing import Optional

from .bridge import MessagingBridge
from .types import AgentAdapter, ServeOptions


class _JsonFormatter(logging.Formatter):
    def format(self, record: logging.LogRecord) -> str:
        return json.dumps({"level": record.levelname.lower(), "msg": record.getMessage()})


def serve(adapter: AgentAdapter, options: Optional[ServeOptions] = None) -> None:
    """Connect an agent adapter to the Astro messaging service and start listening.

    Blocks until the process receives SIGINT or SIGTERM.
    """
    handler = logging.StreamHandler(sys.stdout)
    handler.setFormatter(_JsonFormatter())
    logging.basicConfig(level=logging.INFO, handlers=[handler], force=True)
    bridge = MessagingBridge(adapter, options)
    try:
        asyncio.run(bridge.start())
    except Exception as e:
        print(f"Fatal error: {e}", file=sys.stderr)
        sys.exit(1)
